tango/__main__.py

```diff
@@ -612,9 +641,18 @@ def _run(
         workspace = default_workspace
 
     # Initialize step graph and register run.
-    step_graph = StepGraph(params.pop("steps", keep_as_dict=True))
+    step_graph = StepGraph.from_params(params.pop("steps", keep_as_dict=True))
     params.assert_empty("'tango run'")
-    run = workspace.register_run(step for step in step_graph.values())
+
+    if step_name is not None:
+        assert step_name in step_graph, (
+            f"You want to run a step called '{step_name}', but it cannot be found in the experiment config. "
+            f"The config contains: {list(step_graph.keys())}."
+        )
+        sub_graph = step_graph.get_sub_graph(step_name)
+        run = workspace.register_run(step for step in sub_graph.values())
```

Comment:
Does this mean that a new run will be created for each step that `MulticoreExecutor` runs?

Reply:
If so, I think we want to avoid that. That would definitely create issues with the `WandbWorkspace`.

Reply:
Yes, this is on my TODO to fix. Ideally, we want a single registered run, and the other processes "editing" that run.